x-pack/plugins/security_solution/public/cloud_security_posture/components/vulnerabilities/vulnerabilities_preview.tsx

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { css } from '@emotion/react';
+import numeral from '@elastic/numeral';
+import type { EuiThemeComputed } from '@elastic/eui';
+import { EuiFlexGroup, EuiFlexItem, EuiSpacer, EuiText, useEuiTheme, EuiTitle } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { DistributionBar } from '@kbn/security-solution-distribution-bar';
+import { useVulnerabilitiesPreview } from '@kbn/cloud-security-posture/src/hooks/use_vulnerabilities_preview';
+import { euiThemeVars } from '@kbn/ui-theme';
+import { i18n } from '@kbn/i18n';
+import { ExpandablePanel } from '@kbn/security-solution-common';
+import { buildEntityFlyoutPreviewQuery } from '@kbn/cloud-security-posture-common';
+
+export const getAbbreviatedNumber = (value: number) => {
+  if (isNaN(value)) {
+    return 0;
+  }
+  return value < 1000 ? value : numeral(value).format('0.0a');
+};
+
+interface VulnerabilitiesDistributionBarProps {
+  key: string;
+  count: number;
+  color: string;
+}
+
+const getVulnerabilityStats = (
+  critical: number,
+  high: number,
+  medium: number,
+  low: number,
+  none: number
+): VulnerabilitiesDistributionBarProps[] => {
+  const vulnPropsArray: VulnerabilitiesDistributionBarProps[] = [];
+  if (critical === 0 && high === 0 && medium === 0 && low === 0 && none === 0)
+    return vulnPropsArray;
+
+  if (none > 0)
+    vulnPropsArray.push({
+      key: i18n.translate(
+        'xpack.securitySolution.flyout.right.insights.vulnerabilities.noneVulnerabilitiesText',
+        {
+          defaultMessage: 'None',
+        }
+      ),
+      count: none,
+      color: '#aaa',
+    });
+  if (low > 0)
+    vulnPropsArray.push({
+      key: i18n.translate(
+        'xpack.securitySolution.flyout.right.insights.vulnerabilities.lowVulnerabilitiesText',
+        {
+          defaultMessage: 'Low',
+        }
+      ),
+      count: low,
+      color: euiThemeVars.euiColorVis0,
```

Comment:
Ideally we should use a central function for these colors. 

We currently have this function `getSeverityStatusColor` [here](https://github.com/elastic/kibana/blob/883dfa8ae887c1ee57049a45d8ed8ceb7c2a34d6/x-pack/plugins/cloud_security_posture/public/common/utils/get_vulnerability_colors.ts#L24). 

I think we should move it to the common package to be reusable by all the packages so we avoid those values being out of sync.

Reply:
ooo 
I was planning to move that function on the next PR (as we will use that for the table) but you are right we can move it now and use it here :D